src/lavinmq/exchange/mqtt.cr

```diff
@@ -42,7 +42,7 @@ module LavinMQ
         @retain_store.retain(topic, msg.body_io, msg.bodysize)
       end
 
-      @tree.each_entry(topic) do |queue, qos|
+      @tree.each_entry(msg.routing_key) do |queue, qos|
```

Comment:
How come we pass a routing key here? Seems wrong.

Reply:
we store the bindings as routing keys i think

Reply:
but i got to admit, this routing key to topic stuff is confusing so i might be off somewhere

Reply:
Yeah, it's confusing. 
Should be topics in the tree, not routing keys.

Reply:
wdyt, put time into fixing that or start implementing the native solution so we don't need the conversion everywhere?